### ISSUE TITLE
rust-toolchain causing an error while compiling packages

### ISSUE THREAD

**Issue by jromwu** [rust-corpus/qrates#18]
Seems like a wrong `rustup` command is executed while compiling the packages, here is the error message:
```
01:14:41 [ INFO] installing toolchain [toolchain]
channel = "nightly-2021-01-01"components = [ "rustfmt", "rustc-dev", "llvm-tools-preview" ]01:14:41 [ INFO] running `"/home/jwu/rust-corpus/workspace/cargo-home/bin/rustup" "toolchain" "install" "[toolchain]\nchannel = \"nightly-2021-01-01\"\ncomponents = [ \"rustfmt\", \"rustc-dev\", \"llvm-tools-preview\" ]" "--profile" "minimal"`
01:14:41 [ INFO] [stderr] error: invalid toolchain name: '[toolchain]
01:14:41 [ INFO] [stderr] channel = "nightly-2021-01-01"
01:14:41 [ INFO] [stderr] components = [ "rustfmt", "rustc-dev", "llvm-tools-preview" ]'
The application panicked (crashed).
Message:  Error: unable to install toolchain [toolchain]
channel = "nightly-2021-01-01"
components = [ "rustfmt", "rustc-dev", "llvm-tools-preview" ] via rustup, None
Location: manager/src/lib.rs:66
```

Changing the content of `./rust-toolchain` to just `nightly-2021-01-01` seems to solve the problem. 

The command I used is `sudo env "PATH=$PATH" cargo run --release -- compile`, as `sudo $(which cargo) run --release -- compile` cannot find `rustc`. I am using Ubuntu 20.04 on WSL 2.

**Comment by vakaras** [comment]
The rust-toolchain format we are using requires the latest rustup. Updating it should solve the problem:

```bash
rustup self update
```

If it does not, please let me know. We switched to the format because it allows specifying the required components, which, otherwise, have to be installed manually.

**Comment by jromwu** [comment]
I believe I have the latest version of `rustup`:
```
$ rustup self update
info: checking for self-updates
  rustup unchanged - 1.23.1

$ rustup
rustup 1.23.1 (3df2264a9 2020-11-30)
The Rust toolchain installer

USAGE:
    rustup [FLAGS] [+toolchain] <SUBCOMMAND>

FLAGS:
    -v, --verbose    Enable verbose output
    -q, --quiet      Disable progress output
    -h, --help       Prints help information
    -V, --version    Prints version information

ARGS:
    <+toolchain>    release channel (e.g. +stable) or custom toolchain to set override

SUBCOMMANDS:
    show           Show the active and installed toolchains or profiles
    update         Update Rust toolchains and rustup
    check          Check for updates to Rust toolchains
    default        Set the default toolchain
    toolchain      Modify or query the installed toolchains
    target         Modify a toolchain's supported targets
    component      Modify a toolchain's installed components
    override       Modify directory toolchain overrides
    run            Run a command with an environment configured for a given toolchain
    which          Display which binary will be run for a given command
    doc            Open the documentation for the current toolchain
    man            View the man page for a given command
    self           Modify the rustup installation
    set            Alter rustup settings
    completions    Generate tab-completion scripts for your shell
    help           Prints this message or the help of the given subcommand(s)

DISCUSSION:
    rustup installs The Rust Programming Language from the official
    release channels, enabling you to easily switch between stable,
    beta, and nightly compilers and keep them updated. It makes
    cross-compiling simpler with binary builds of the standard library
    for common platforms.

    If you are new to Rust consider running `rustup doc --book` to
    learn Rust.
```

**Comment by vakaras** [comment]
The problem seems to be that my code parsing the `rust-toolchain` file was outdated. Should be fixed now (#25). Does it work for you?

**Comment by jromwu** [comment]
Yeah it works perfectly for me now. Thanks :)
